### PR TITLE
Taggable shortforms

### PR DESCRIPTION
**Pull request by jpaddison3**
This feature is Forum-gated for now, but is probably of interest to LW.

This PR moves us from a world of subforum discussions being comments on the topic, to being shortform comments that happen to be tagged with the topic. This reflects the general shift we've had away from content that is specific to the subforum, and towards a world where ""subforums"" are a view onto the content of the Forum that surfaces what is interesting to you.  More in [this google doc](https://docs.google.com/document/d/1QplakxW3mUsLsGIUkqAgdBVNFdt4oUFhC-yk4aVr6iI/edit#heading=h.ksyhcb9ezvjy).

The first major choice is that I did not want us to go with full-tag-relevance voting. Instead the topic relevance is determined by the author of the comment. This simplifies the UI and the architecture.

Design-wise, the major tough decision is how to display which topics the shortform is tagged with in the small space of the comment UI. We've gone for an icon approach. It ... did not get universal acclaim when previewed on slack, but I failed to follow up, and haven't seen anything better yet, so I'm opening this PR as-is now. Here it is again:

image

Second design bit, this UI:

image

Product-wise note two things:
 1. I've disabled shortform on the frontpage until we ship a checkbox allowing users to opt-out of their shortform posts being shown on the frontpage.
 2. Shortforms currently have no titles. I think this is fine/good, but I expect @SharangP to be of the opinion that we should add them. I don't expect him to want it in this PR though.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203872041451606) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
Also contains https://github.com/ForumMagnum/ForumMagnum/pull/6613 because I needed it.

**Comment by jpaddison3** [comment]
LW, It is my view that, conceptually, this moves y'all closer to the world of tagging shortforms, however:
1. You don't have anywhere to display those shortforms on a tag yet.
2. On the assumption that you don't add icons to all your core tags, if we do go for the icon approach, you will need to build something like a mini-tag-chip design to replace them.

**Comment by jpaddison3** [comment]
Lol. Me: "I don't have time to review design files, I need to write code to ship this big thing before sprint end." Also me, now: "oh look at this great minified tag chip."

image

**Review by s-cheng — APPROVED** [review]
Nice, really appreciate all the code improvements! I understand that there's some urgency here and I don't think any of the issues are quite blockers, plus I trust you to read and address all my comments, so I'll approve this PR. I would def suggest asking Agnes for a design review, and feel free to ask me to do another code review.

Actually I think there is a blocker, that is that shortforms no longer appear in Recent Discussion, which we don't want to affect LW.

File: packages/lesswrong/components/form-components/FormComponentTagsChecklist.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    limit: 100,
+  });
+
+  const { Loading, TagsChecklist } = Components;
+
+  // TODO: Currently this component is only used in the comment form. Other than
+  // this line, and the line about truncate below, it is fully general, however.
+  if (!document.shortform) {
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/form-components/FormComponentTagsChecklist.tsx:43]: I think this is the way we should leave this until we need to use this for another component. (Agreeing with "But this seems fine for now.")

Later, I think we should have a disabled ----- .... waaaaait a minute, you mean like `hidden`? I had thought of that, but then looked at hidden and thought it didn't have access to the document, except, nope, it totally does.

**Pull request merged**